R/status.R

```diff
@@ -1,10 +1,65 @@
 
 the$status_running <- FALSE
 
-#' Report differences between lockfile and project library
+#' Report inconsistencies between lockfile, library, and dependencies
 #'
-#' Report differences between the project's lockfile and the current state of
-#' the project's library (if any).
+#' `status()` reports issues caused by inconsistencies across the project
+#' lockfile, library, and [dependencies()]. In general, you should strive to
+#' ensure that `status()` reports no issues, as this maximises your chances of
+#' successfully `restore()`ing the project in the future or on another machine.
+#'
+#' See the headings below for specific advice on resolving any issues
+#' revealed by `status()`.
+#'
+#' # Missing packages
+#'
+#' First, ensure that all packages used by the project are installed. This is
+#' important to do first because if any packages are missing we can't tell for
+#' sure that a package isn't used; it might be a dependency that we don’t know
+#' about.
+#'
+#' There are four possibilities for an uninstalled package:
```

Comment:
What should users do when multiple packages are in an inconsistent state? E.g. you have some packages which are used and recorded; and others which are recorded but not used.

(I think we want to recommend using `renv::restore()` first; perhaps `renv::restore(packages = <...>)`, and then using `renv::snapshot()` to remove unused packages?)